pandas-stubs/core/series.pyi

```diff
@@ -676,15 +685,15 @@ class Series(IndexOpsMixin[S1], NDFrame):
     def diff(self, periods: int = ...) -> Series[S1]: ...
     def autocorr(self, lag: int = ...) -> float: ...
     @overload
-    def dot(self, other: Series[S1]) -> Scalar: ...
+    def dot(self, other: Series[S1]) -> Scalar: ...  # type: ignore[misc] # pyright: ignore[reportOverlappingOverload]
     @overload
     def dot(self, other: DataFrame) -> Series[S1]: ...
     @overload
     def dot(self, other: _ListLike) -> np.ndarray: ...
```

Comment:
Maybe use something besides `_ListLike` here to avoid the overlapping overload a few lines earlier?